### PR TITLE
【翻译完成】20130910-Slices-of-Slices-of-Slices-in-Go

### PR DESCRIPTION
**Pull request by xmge**
#743

### CODE REVIEW

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+// Represents a marine station and its polygons
+type MarineStation struct {
+    StationId string      bson:&quot;station_id&quot;
+    Polygons PolygonRings bson:&quot;polygons&quot;
+}
+```
+
+多边形区域结构由长度为2的 float 数组切片组成。切片中每一个数组标示多边形区域结构中的一个点。
+
+如果你要是想通过 mongodb 来执行不同区域的地理空间搜索，那么在 mongodb 中存储多边形区域数据的结构是必须的。
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
如果你要是想通过 mongodb 来执行不同多边形区域的地理空间搜索，那么在 mongodb 中存储多边形区域数据的结构是必须的。
```

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+一个海洋预报区域当成一个与很多多边形区域结构有关系的站。
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
MarineStation 结构体模拟一个单独的站点和对应的多边形区域。
```

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+测试代码将创建一个带有两个多边形区域结构的站。然后它会显示一切。让我们来看看如何用 slice 创建一个海洋站，并创建一个单一的海洋站进行测试：
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
测试代码将创建一个带有两个多边形区域结构的站。然后它会显示一切。让我们来看看如何用切片创建一个海洋站，并创建一个单一的海洋站进行测试：
```

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    Polygons: PolygonRings{
+        Type: "Polygon",
+        Coordinates: []Polygon{},
+    },
+}
+```
+
+第一行代码创建了一个存储海洋预报区域的空切片。然后我们用复合字面量的方式的方式创建了一个海洋预报区域对象。在这个复合字面量中我们需要为这个 gePolygonRings 创建另一个复合字面量对象 Polygons 。在创建的 PolygonRings 对象中我们为 Coordinates 字段创建空的 slice 来存储 Polygon 对象。
+
+若要了解复合文本的更多信息，请查看此文档：
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
若要了解复合字面量的更多信息，请查看此文档：
```

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+marineStation := MarineStation{
+    StationId: "AMZ123",
+    Polygons: PolygonRings{
+        Type: "Polygon",
+        Coordinates: []Polygon{},
+    },
+}
+```
+
+第一行代码创建了一个存储海洋预报区域的空切片。然后我们用复合字面量的方式的方式创建了一个海洋预报区域对象。在这个复合字面量中我们需要为这个 gePolygonRings 创建另一个复合字面量对象 Polygons 。在创建的 PolygonRings 对象中我们为 Coordinates 字段创建空的 slice 来存储 Polygon 对象。
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
第一行代码创建了一个存储 MarineStation 对象的空切片。然后我们用复合字面量的方式，创建了一个 MarineStation 对象。在这个复合字面量中，我们需要为这个 PolygonRings，创建另一个复合字面量对象 Polygons。在创建的 PolygonRings 对象中，我们为 Coordinates 字段创建空的切片来存储 Polygon 对象。
```

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+在第二个 polygon 中有4个点而不是5个，剩下的最后一件事就是将 polygon 加入到 stations 切片中，并且展示出来：
+
+```go
+// Add the marine station
+marineStations = append(marineStations, marineStation)
+
+Display(marineStations)
+```
+
+展示函数使用关键字`range`来进行遍历所有的切片。
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
Display 函数使用关键字 `range` 来进行遍历所有的切片。
```

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+            for _, coordinate := range rings {
+                fmt.Printf("Point: %f,%f\n", coordinate[0], coordinate[1])
+            }
+        }
+    }
+}
+```
+
+这个方法需要传入一个 MarineStation 切片作为参数。记住 slice 结构仅仅是在栈上被拷贝，并不是传入 slice 对象。
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
这个方法需要传入一个 MarineStation 切片作为参数。记住，在栈上拷贝的仅仅是切片的结构，而不是切片表示的所有对象。
```

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+Point: -80.437622,27.788513
+Point: -80.438416,27.788513
+Point: -80.437012,27.787720
+```
+
+使用slice去解决这个问题是快速的、容易的、高效的。我将这份测试代码复制了一份放在了[The Go Playground](https://play.golang.org/)。
+
+http://play.golang.org/p/UYO2HIKggy
+
+通过快速构建这个测试应用让我深深地感觉到 slice 具有很大的优点。它可以使你开发更高效，代码更健壮。你不必担心内存管理,你可以通过切片的引用在函数数据传递时传递较大的数据。花一些时间去学习如何在代码中使用 slice，学完后你将感谢你自己。
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
通过快速构建这个测试应用，让我深深地感觉到切片具有很大的优点。它可以使你开发更高效，代码更健壮。你不必担心内存管理，你可以通过切片的引用，在函数数据传递时传递较大的数据。花一些时间去学习在代码中使用切片，你会很有收获。
```

File: translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+---
+
+via: https://www.ardanlabs.com/blog/2013/09/slices-of-slices-of-slices-in-go.html
+
+作者：[William Kennedy](https://github.com/ardanlabs/gotraining)
+译者：[xmge](https://github.com/xmge)
+校对：[polaris1119](https://github.com/polaris1119)
```
Comment (Noluye) [inline translated/tech/20130910-Slices-of-Slices-of-Slices-in-Go.md]:
```suggestion
校对：[Noluye](https://github.com/Noluye)
```

**Comment by xmge** [comment]
ok,辛苦啦 👍

**Pull request merged**